crates/codegen/runtime/cargo/src/runtime/diagnostic.rs

```diff
@@ -0,0 +1,65 @@
+use std::borrow::Cow;
+use std::error::Error;
+
+use crate::text_index::TextRange;
+
+#[repr(u8)]
+pub enum Severity {
+    Error = 1,
+    Warning = 2,
+    Information = 3,
+    Hint = 4,
+}
+
+pub trait Diagnostic: Error {
```

Comment:
We don't strictly have to rely on `Error` bound here, I just used it to have the `Error: Display` in the hierarchy and use that impl (`Error::description` is obsolete in favor of implementing Error and using `Display`).

Let me know and I can simply rely on `Display` here or don't specify any bounds and leave it up to implementors to correctly render the message themselves.

Reply:
After I wrote it, it occurred to me that we probably will not want to always enforce on the `Error` bound (e.g. warnings), so I dropped it in https://github.com/NomicFoundation/slang/pull/963/commits/89bc4718842b4ac48887e1ef6efc374e71472306.